dapos/dapos_handshake.go

```diff
@@ -518,25 +517,33 @@ func executeTransaction(transaction *types.Transaction, receipt *types.Receipt,
 			return
 		}
 
-
 		dvmService := dvm.GetDVMService()
 		dvmResult, err1 := dvmService.ExecuteSmartContract(transaction)
 		if err1 != nil {
 			utils.Error(err, utils.GetCallStackWithFileAndLineNumber())
 		}
 
+		hertz = minHertzUsed + dvmResult.CumulativeHertzUsed
+
 		err = processDVMResult(transaction, dvmResult, receipt)
 		if err != nil {
 			utils.Error(err)
+
 			receipt.Status = types.StatusInternalError
 			receipt.HumanReadableStatus = err.Error()
 			receipt.Cache(services.GetCache())
+
+			rateLimit, err := types.NewRateLimit(transaction.From, transaction.Hash,  hertz)
```

Comment:
@restevens402 is transaction.From going away? Or do we populate it automatically when we unravel the signature?

Reply:
right now it is part of the original submitted tx.  We have proposed the plan of eliminating that and using the value derived from the signature.